iphone/Maps/Core/iCloud/SynchronizationFileWriter.swift

```diff
@@ -128,8 +128,13 @@ final class SynchronizationFileWriter {
 
   private func removeFromCloudContainer(_ localMetadataItem: LocalMetadataItem, completion: @escaping WritingResultCompletionHandler) {
     LOG(.info, "Start trashing file \(localMetadataItem.fileName)...")
+    let targetCloudFileUrl = localMetadataItem.relatedCloudItemUrl(to: cloudDirectoryUrl)
```

Comment:
How is it possible that the removed file does not exist in the iCloud?

Reply:
The file removing from the cloud isn't synchronous operation and require some time.

If you remove the file, than fast  press recover and delete again, the file may be already deleted in the cloud directory and get a new action to be deleted.

Reply:
Shouldn't the recovered file be immediately copied back into the iCloud folder? It looks like a bug in our logic that relies on local and iCloud folders being always in sync. Copying files locally should be instant, no?

The referenced issue says:

> but the deletion from the cloud container wasn't so fast and still not perform the moving to the trash dir.

It describes another case when a file _is still present_ in the iCloud folder, and not deleted yet.


While the logic to not crash when a non-existing file is deleted looks obvious, the main question is "How is it possible that we're deleting a file that we think should exist, but it doesn't?". It would be great to make sure that we are _fixing_ the root cause here, and not hiding it.